### PR TITLE
SCC-3745 - Add jest and react testing library

### PR DESCRIPTION
**Pull request by dgcohen**
This PR adds Jest and React Testing Library as well as the initial config.

The latest version of RTL requires React 18 so I installed version 12.1.5 for now which compatible with React 17.

I also added 2 basic tests for testing the initial config but there is currently no minimum coverage required, as well as a github workflow to run the unit test suite on code pushes.

NB: Tests for the page components have to be placed in the __test__ directory in the root app, while component tests can be located next to the components in the src directory.

### CODE REVIEW

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Some updates. I personally prefer colocating the test file with the component file but that's a good question to bring up with the team.

**Review by charmingduchess — APPROVED** [review]

File: .husky/pre-commit [hunk 1]
```diff
@@ -3,3 +3,4 @@
 
 npm run prettier
 npm run lint
+npm run test
```
[the remark this answers is not in the document]
Comment (dgcohen) [reply .husky/pre-commit:6]: Good call, happy to get rid of this also since it can be annoying.

**Comment by dgcohen** [comment]
> Some updates. I personally prefer colocating the test file with the component file but that's a good question to bring up with the team.

Yeah I was thinking of doing that initially, might be nice to have them live near the components.

**Review by EdwinGuzman — APPROVED** [review]
